### ISSUE TITLE
closest's second argument doesn't appear to work

### ISSUE THREAD

**Issue by aaronjensen** [facebook/jscodeshift#41]
Here's an example: http://felix-kling.de/esprima_ast_explorer/#/WhIlcdMzwh/1

```js
function foo() {
  function bar() {
    return 3;
  }
}
```

```js
module.exports = function(file, api) {
  var j = api.jscodeshift;

  return j(file.source)
    .find(j.Literal)
    .forEach(function(path) { console.log(path.value) })
    .closest(j.FunctionDeclaration, { id: { name: "foo" } })
    .forEach(function(path) { 
      console.log(path.value);
      path.value.id.name = 'renamed'; })
    .toSource();
};
```

I'd expect this to work like find does, using the 2nd parameter as a filter. Unfortunately, adding the 2nd parameter causes it to find the `ReturnStatement` instead of the top level Foo function

**Comment by aaronjensen** [comment]
Nice, thanks!

**Comment by fkling** [comment]
I'm trying to tackle another bug first before I release a new version. I will definitely make a new release tomorrow though.

**Comment by fkling** [comment]
Just released v0.3.6 with the fix.
